maraboupy/runMarabou.py

```diff
@@ -0,0 +1,219 @@
+import numpy as np
+import os
+import sys
+
+import pathlib
+sys.path.insert(0, os.path.join(str(pathlib.Path(__file__).parent.absolute()), "../"))
+from maraboupy import Marabou
+from maraboupy import MarabouCore
+from maraboupy import MarabouUtils
+
+def main():
+        args = arguments().parse_args()
+        print(args)
+        query, network = createQuery(args)
+        if (args.query_dump_file != ""):
+                MarabouCore.saveQuery(query, args.query_dump_file)
+                exit()
+        if query == None:
+            print("Unable to create an input query!")
+            print("There are three options to define the benchmark:\n"
+                  "1. Provide an input query file.\n"
+                  "2. Provide a network and a property file.\n"
+                  "3. Provide a network, a dataset (--dataset), an epsilon (-e), "
+                  "target label (-t), and the index of the point in the test set (-i).")
+            exit(1)
+        options = createOptions(args)
+        vals, stats = MarabouCore.solve(query, options)
+        if stats.hasTimedOut():
+            print ("TIMEOUT")
+        elif len(vals)==0:
+            print("unsat")
+        else:
+            for i in range(query.getNumInputVariables()):
+                print("x{} = {}".format(i, vals[query.inputVariableByIndex(i)]))
+            if network:
+                outputs = network.evaluateWithoutMarabou(np.array([[vals[query.inputVariableByIndex(i)] for i in range(query.getNumInputVariables())]]))
+                for i, output in enumerate(list(outputs.flatten())):
+                    print("y{} = {}".format(i, output))
+            else:
+                for i in range(query.getNumOutputVariables()):
+                    print("y{} = {}".format(i, vals[query.outputVariableByIndex(i)]))
+
+            print("sat")
+
+def createQuery(args):
+    if args.input_query:
+        query = Marabou.load_query(args.input_query)
+        return query, None
+    networkPath = args.network
+
+    suffix = networkPath.split('.')[-1]
+    if suffix == "nnet":
+        network = Marabou.read_nnet(networkPath)
+    elif suffix == "pb":
+        network = Marabou.read_tf(networkPath)
+    elif suffix == "onnx":
+        network = Marabou.read_onnx(networkPath)
+    else:
+        print("The network must be in .pb, .nnet, or .onnx format!")
+        return None, None
+
+    if  args.prop != None:
+        query = network.getMarabouQuery()
+        if MarabouCore.loadProperty(query, args.prop):
+            return query, network
+
+    if args.dataset == 'mnist':
+        encode_mnist_linf(network, args.index, args.epsilon, args.target_label)
+        return network.getMarabouQuery(), network
+    elif args.dataset == 'cifar10':
+        encode_cifar10_linf(network, args.index, args.epsilon, args.target_label)
+        return network.getMarabouQuery(), network
+    else:
+        print("No property encoded! The dataset must be taxi or mnist or cifar10.")
```

Comment:
I didnt understand the comment. Can you have both a dataset and a property? I guess not